src/Riok.Mapperly/Emit/Syntax/SyntaxFactoryHelper.GeneratedCode.cs

```diff
@@ -0,0 +1,26 @@
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Riok.Mapperly.Emit.Syntax;
+
+public partial struct SyntaxFactoryHelper
+{
+    private const string GeneratedCodeAttributeName = "global::System.CodeDom.Compiler.GeneratedCode";
+    private static readonly AssemblyName _generatorAssemblyName = typeof(SyntaxFactoryHelper).Assembly.GetName();
+
+    public SyntaxList<AttributeListSyntax> GeneratedCodeAttributeList()
+    {
+        var arguments = CommaSeparatedList(
+            AttributeArgument(LiteralExpression(SyntaxKind.StringLiteralExpression, Literal(_generatorAssemblyName.Name))),
+            AttributeArgument(LiteralExpression(SyntaxKind.StringLiteralExpression, Literal(_generatorAssemblyName.Version.ToString())))
+        );
+
+        var attribute = Attribute(IdentifierName(GeneratedCodeAttributeName))
+            .WithArgumentList(AttributeArgumentList(CommaSeparatedList(arguments)));
+
+        return SingletonList(AttributeList(SingletonSeparatedList(attribute)).AddTrailingLineFeed(Indentation));
```

Comment:
I think it would make sense to create a new helper method in `SyntaxFactoryHelper.Attribute.cs` which creates the attribute list (e.g. 
```c#
SyntaxList<AttributeListSyntax> AttributeList(string name, params Expression[] arguments)
{
    var arguments = CommaSeparatedList(arguments.Select(x => new AttributeArgument(x));

    var attribute = Attribute(IdentifierName(name)).WithArgumentList(AttributeArgumentList(CommaSeparatedList(arguments)));

    return SingletonList(AttributeList(SingletonSeparatedList(attribute)).AddTrailingLineFeed(Indentation));
}
```

This could also be used for `SyntaxFactoryHelper.UnsafeAccessorAttributeList`.